### PR TITLE
MAINT: Migrate testdata to nipreps-data and remove docker-wrapper

### PR DESCRIPTION
**Pull request by oesteban**
Initial refactor of CircleCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -104,28 +103,54 @@ jobs:
...
-            - data-v0-{{ .Branch }}-{{ .Revision }}
-            - data-v0-{{ .Branch }}
-            - data-v0-
+            - env-v0-{{ .Branch }}-
+            - env-v0-master-
+            - env-v0-
+      - run:
+          name: Setup git-annex & DataLad
+          command: |
+            conda install -y -c anaconda -c conda-forge git-annex datalad
```
Comment (mgxd) [inline .circleci/config.yml:120]: is there a need to add the anaconda channel if both of those are on conda-forge?
```suggestion
            conda install -y -c conda-forge git-annex datalad
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -104,28 +103,54 @@ jobs:
...
-             rm -rf /tmp/data/rodent-se/
-             wget --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 0 -q \
-               -O rodent-se.tar.gz "https://stanford.box.com/shared/static/ei6qtu81y1zx9fxii11svvm73n3884fl.gz"
-             tar xvzf rodent-se.tar.gz -C /tmp/data/
+          name: Setup TemplateFlow
+          command: |
+            python -m pip install --no-cache-dir -U templateflow
+            python -c "from templateflow import api as tfapi; \
+                       tfapi.get('OASIS30ANTs', resolution=1, desc=None, suffix='T1w'); \
+                       tfapi.get('OASIS30ANTs', resolution=1, desc='BrainCerebellumExtraction', suffix='mask'); "
```
Comment (mgxd) [inline .circleci/config.yml:130]: we're not using the OASIS template here
```suggestion
                       tfapi.get('Fischer344', desc=None, suffix='T2w'); \
                       tfapi.get('Fischer344', desc='BrainCerebellumExtraction', suffix='mask'); "
```

**Review by mgxd — COMMENTED** [review]
+1 on moving towards a datalad workflow for data storage / retrieval. Few comments

**Review by eilidhmacnicol — COMMENTED** [review]
Looks fine to me, but I'm not well versed in circleCI yet. 

When I've been testing the full workflow, I've needed `--fs-no-reconall` and `--force-no-bbr` but as we're planning on taking these out, then I suppose there's not much use in adding them at this stage?

File: .circleci/config.yml (hunk 1 above)
Comment (oesteban) [reply .circleci/config.yml:120]: AFAIK git-annex is not on conda-forge

**Comment by oesteban** [comment]
Let's go for it

File: .circleci/config.yml (hunk 1 above)
Comment (mgxd) [reply .circleci/config.yml:120]: FWIW: https://github.com/conda-forge/git-annex-feedstock

**Pull request merged**